### PR TITLE
Convert PSXXXX to ParseStatementFlags

### PR DESCRIPTION
**Pull request by wilzbach**
```bash
replacements=(
"PSscope scope_"
"PScurlyscope curlyScope"
"PSsemi_ok semiOk"
"PScurly curly"
"PSsemi semi"
)
shopt -s globstar
for r in "${replacements[@]}" ; do
    w=($r)
    sed "s/${w[0]}/ParseStatementFlags.${w[1]}/g" -i **/*.d
done
```

### CODE REVIEW

File: src/dmd/parse.d [hunk 1]
```diff
@@ -193,19 +193,13 @@ __gshared PREC[TOKMAX] precedence =
...
 {
-    PSsemi          = 1,        // empty ';' statements are allowed, but deprecated
-    PSscope         = 2,        // start a new scope
-    PScurly         = 4,        // { } statement is required
-    PScurlyscope    = 8,        // { } starts a new scope
-    PSsemi_ok       = 0x10,     // empty ';' are really ok
+    semi          = 1,        // empty ';' statements are allowed, but deprecated
+    scope_        = 2,        // start a new scope
+    curly         = 4,        // { } statement is required
+    curlyscope    = 8,        // { } starts a new scope
```
Comment (JinShil) [inline src/dmd/parse.d]: We should consider if we want to enforce the style guide here and make it `curlyScope`.

Comment (wilzbach) [reply src/dmd/parse.d]: It would also be `semiOk` ...

Comment (JinShil) [reply src/dmd/parse.d]: Yes.

Comment (wilzbach) [reply src/dmd/parse.d]: Ok changed.

Comment (jacob-carlborg) [reply src/dmd/parse.d]: Not that I'm complaining, but we haven't done that in any other of these PRs. All enum names are still all uppercase.

**Review by jacob-carlborg — APPROVED** [review]

File: src/dmd/parse.d [hunk 2]
```diff
@@ -2640,7 +2634,7 @@ final class Parser(AST) : Lexer
             check(TOKrparen);
         }
 
-        auto fbody = parseStatement(PScurly);
+        auto fbody = parseStatement(ParseStatementFlags.curly);
         auto f = new AST.InvariantDeclaration(loc, token.loc, stc, null, fbody);
```
Comment (ZombineDev) [inline src/dmd/parse.d]: `ParseStatementFlags` is quite long. How about adding a private `alias` at the top of the module:
```D
private alias PS = ParseStatementFlags;
//...
auto fbody = parseStatement(PS.curly);
```

Comment (jacob-carlborg) [reply src/dmd/parse.d]: I disagree, it's not used *that* much.

Comment (JinShil) [reply src/dmd/parse.d]: I'm OK with or without the alias.

**Review by JinShil — APPROVED** [review]

**Pull request merged**